### PR TITLE
Add interface for 2p imaging data, suite2p and spike2 signals in dual acquisition modality

### PR DESCRIPTION
**Pull request by alessandratrapani**
Small PR to add interfaces as in the 2p-only configuration https://github.com/catalystneuro/higley-lab-to-nwb/pull/22
- [x] add interface for 2p data in dual modality (probably just reusing the one for 2p only acquisition) 
- [x] add suuite2p interface
- [x] add spike 2 signals
- [x] add correct metadata for dual acquisition
- [x] time alignment

PS: no behavioral video present. Facemap output will be added in a followup PR after solving #19

### CODE REVIEW

**Comment by alessandratrapani** [comment]
stub nwb file [here](https://catalystneuro-processing.s3.us-east-2.amazonaws.com/higley-lab-to-nwb/20201231_00001_dbvdual035.nwb)

File: src/higley_lab_to_nwb/benisty_2024/metadata/benisty_2024_dual_ophys_metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      excitation_lambda: 920.0  # in nm
+      location: Whole Brain
+      device: TwoPhotonMicroscope
+      optical_channel:
+        - name: Green
+          description: Emitted light was collected through a 525/50 filter and a gallium arsenide phosphide photomultiplier tube (Hamamatsu).
+          emission_lambda: 525.0
+      indicator: GCaMP6s #TODO change
+    - name: OnePhotonImagingPlaneRedChannel
+      description: Imaging plane for the Green channel recorded with 2p microscope.
```
Comment (weiglszonja) [inline src/higley_lab_to_nwb/benisty_2024/metadata/benisty_2024_dual_ophys_metadata.yaml:39]: is it not 1p microscope then?

**Comment by weiglszonja** [comment]
The wheel signal here has [100, 1] dimension, while in the other example it was [100], I think it should be [100] here as well, is it not?

**Comment by weiglszonja** [comment]
The `OnePhotonMicroscope` is added here, but there is not `ImagingPlane` linked to it, is it intentional?

File: src/higley_lab_to_nwb/benisty_2024/metadata/benisty_2024_dual_ophys_metadata.yaml (hunk 1 above)
Comment (alessandratrapani) [reply src/higley_lab_to_nwb/benisty_2024/metadata/benisty_2024_dual_ophys_metadata.yaml]: Good Catch! NO it should be 1p, I corrected it

**Comment by alessandratrapani** [comment]
> The `OnePhotonMicroscope` is added here, but there is not `ImagingPlane` linked to it, is it intentional?

Yes, I created already the complete metadata yaml, I am going to add the one photo series in a next PR

**Review by weiglszonja — APPROVED** [review]
Ok, looks good!

**Comment by alessandratrapani** [comment]
> The wheel signal here has [100, 1] dimension, while in the other example it was [100], I think it should be [100] here as well, is it not?

That's weird, I re-run it and now they both have shape (100,1)

**Comment by weiglszonja** [comment]
> > The wheel signal here has [100, 1] dimension, while in the other example it was [100], I think it should be [100] here as well, is it not?
> 
> That's weird, I re-run it and now they both have shape (100,1)

In that case I think we should store as (100) if it's a single trace

**Comment by alessandratrapani** [comment]
> > > The wheel signal here has [100, 1] dimension, while in the other example it was [100], I think it should be [100] here as well, is it not?
> > 
> > 
> > That's weird, I re-run it and now they both have shape (100,1)
> 
> In that case I think we should store as (100,) if it's a single trace, but can also discuss in a follow-up

The thing is I am using the neuroconv extractor for this signal (CedRecordingExtractor)

**Comment by weiglszonja** [comment]
> > > > The wheel signal here has [100, 1] dimension, while in the other example it was [100], I think it should be [100] here as well, is it not?
> > > 
> > > 
> > > That's weird, I re-run it and now they both have shape (100,1)
> > 
> > 
> > In that case I think we should store as (100,) if it's a single trace, but can also discuss in a follow-up
> 
> Yes the interface is custom but the extractor is from neuroconv (CedRecordingExtractor)

Yeah I would squeeze the last dimension if this is a single trace. The extractor is from spikeinterface, the interface would be from NeuroConv

**Comment by weiglszonja** [comment]
> > Yeah I would squeeze the last dimension if this is a single trace. The extractor is from spikeinterface, the interface would be from NeuroConv
> 
> Yes, sorry, I meant the interface is ad hoc while the extractor is the CedRecordingExtractor from spikeinterface. So, should I also implement an ad hoc extractor?

No, you can just edit this line here: https://github.com/catalystneuro/higley-lab-to-nwb/blob/ede7e661c17a8c26810daddcf117cb68df90c343/src/higley_lab_to_nwb/lohani_2022/interfaces/lohani_2022_spike2signals_interface.py#L133

**Comment by alessandratrapani** [comment]
> No, you can just edit this line here:
> 
> https://github.com/catalystneuro/higley-lab-to-nwb/blob/ede7e661c17a8c26810daddcf117cb68df90c343/src/higley_lab_to_nwb/lohani_2022/interfaces/lohani_2022_spike2signals_interface.py#L133

Ah yes, thanks! I will do it here #23 since this affect both conversions

**Comment by weiglszonja** [comment]
okok, good to go then

**Pull request merged**